File: app/src/main/java/com/platdmit/simplecloudmanager/data/api/models/ApiAction.java
package com.platdmit.simplecloudmanager.data.api.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class ApiAction {
    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("status")
    @Expose
    private String status;
    @SerializedName("type")
    @Expose
    private String type;
    @SerializedName("started_at")
    @Expose
    private String startedAt;
    @SerializedName("completed_at")
    @Expose
    private String completedAt;
    @SerializedName("resource_id")
    @Expose
    private String resourceId;
    @SerializedName("resource_type")
    @Expose
    private String resourceType;
    @SerializedName("initiator")
    @Expose
    private String initiator;
    @SerializedName("region_slug")
    @Expose
    private String regionSlug;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getStartedAt() {
        return startedAt;
    }

    public void setStartedAt(String startedAt) {
        this.startedAt = startedAt;
    }

    public String getCompletedAt() {
        return completedAt;
    }

    public void setCompletedAt(String completedAt) {
        this.completedAt = completedAt;
    }

    public String getResourceId() {
        return resourceId;
    }

    public void setResourceId(String resourceId) {
        this.resourceId = resourceId;
    }

    public String getResourceType() {
        return resourceType;
    }

    public void setResourceType(String resourceType) {
        this.resourceType = resourceType;
    }

    public String getInitiator() {
        return initiator;
    }

    public void setInitiator(String initiator) {
        this.initiator = initiator;
    }

    public String getRegionSlug() {
        return regionSlug;
    }

    public void setRegionSlug(String regionSlug) {
        this.regionSlug = regionSlug;
    }
}

File: mod_servers/src/main/java/com/platdmit/mod_servers/data/retrofit/models/ApiAction.kt
package com.platdmit.mod_servers.data.retrofit.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class ApiAction(
        @SerializedName("id")
        @Expose
        val id: String,

        @SerializedName("status")
        @Expose
        val status: String,

        @SerializedName("type")
        @Expose
        val type: String,

        @SerializedName("started_at")
        @Expose
        val startedAt: String,

        @SerializedName("completed_at")
        @Expose
        val completedAt: String,

        @SerializedName("resource_id")
        @Expose
        val resourceId: String,

        @SerializedName("resource_type")
        @Expose
        val resourceType: String,

        @SerializedName("initiator")
        @Expose
        val initiator: String,

        @SerializedName("region_slug")
        @Expose
        val regionSlug: String
)